### ISSUE TITLE
Error when used with native classes

### ISSUE THREAD

**Issue by ssutar** [machty/ember-concurrency#261]
When used with `ember-native-class-polyfill` and `ember-decorators`, following error is thrown
```
Uncaught TypeError: Class constructor ComputedProperty cannot be invoked without 'new'
    at new TaskProperty (-task-property.js:398)
    at task (index.js:58)
    at Module.callback (search-input.js:43)
    at Module.exports (loader.js:106)
    at requireModule (loader.js:27)
    at Class._extractDefaultExport (index.js:422)
    at Class.resolveOther (index.js:101)
    at Class.resolve (index.js:163)
    at resolve (container.js:970)
    at Registry.resolve (container.js:600)
```
Its originated from https://github.com/machty/ember-concurrency/blob/master/addon/-task-property.js#L418 

cc: @rwjblue

**Comment by rwjblue** [comment]
I think we could likely migrate to using native classes directly.

Also, to be clear, this will be an issue with Ember 3.6.0+ (not just with the polyfill).

**Comment by buschtoens** [comment]
Shameless plug: [ember-concurrency-decorators](https://github.com/machty/ember-concurrency-decorators)

**Comment by toovy** [comment]
@buschtoens fyi also happens when using ember-concurrency-decorators with the ember-native-class-polyfill

**Comment by machty** [comment]
Fix released on 0.8.24
